### DISCUSSION TITLE
Combining one-to-one and one-to-many

### DISCUSSION THREAD

**Discussion by bexuma** [prisma/prisma#12171] [Q&A] [answered]
I want to add `currentVersion RecordHistory` to Record which will return RecordHistory matching Record's id and history values.
```prisma
model Record {
  id       String          @id @default(uuid())
  history  Int             @default(0)
  versions RecordHistory[]
}

model RecordHistory {
  recordId      String
  recordHistory Int      @default(0)
  createdAt     DateTime @default(now())
  record        Record   @relation(fields: [recordId], references: [id], onDelete: Cascade)

  @@id([recordId, recordHistory])
}
```

**Comment by bexuma** [comment, marked as answer]
I have thought of this eventually. Gonna test and report back here.
```prisma
model Record {
  id       String          @id @default(uuid())
  history  Int             @default(0)
  versions RecordHistory[] @relation(name: "versions")
  version  RecordHistory   @relation(name: "version", fields: [id, history], references: [recordId, recordHistory])
}

model RecordHistory {
  recordId      String
  recordHistory Int      @default(0)
  createdAt     DateTime @default(now())
  record        Record   @relation("versions", fields: [recordId], references: [id])
  Record        Record[] @relation("version")

  @@id([recordId, recordHistory])
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
